contracts/PrecompileManager.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./MerkleLib.sol";
+import "./IStorageManager.sol";
+
+contract PrecompileManager is ISystemContractDaggerHashimoto {
+    address public constant systemDeployer = 0x0000000000000000000000000000000000033301;
+    address public constant sstoragePisaPutRaw = 0x0000000000000000000000000000000000033302;
+    address public constant sstoragePisaGetRaw = 0x0000000000000000000000000000000000033303;
+    address public constant sstoragePisaUnmaskDaggerData = 0x0000000000000000000000000000000000033304;
+    address public constant sstoragePisaRemoveRaw = 0x0000000000000000000000000000000000033305;
+
+    // Get a raw data from underlying storage.
+    function getRaw(
+        bytes32 hash,
+        uint256 kvIdx,
+        uint256 off,
+        uint256 len
+    ) public view virtual returns (bytes memory) {
+        (bool success, bytes memory data) = address(sstoragePisaGetRaw).staticcall(abi.encode(hash, kvIdx, off, len));
+        require(success, "failed to getRaw");
+        return data;
+    }
+
+    // Set a raw data to underlying storage.
+    function putRaw(uint256 kvIdx, bytes memory data) public virtual {
+        (bool success, ) = address(sstoragePisaPutRaw).call(abi.encode(kvIdx, data));
+        require(success, "failed to putRaw");
+    }
+
+    // Remove by moving data from fromKvIdx to toKvIdx and clear fromKvIdx
+    function removeRaw(uint256 fromKvIdx, uint256 toKvIdx) public virtual {
+        (bool success, ) = address(sstoragePisaRemoveRaw).call(abi.encode(fromKvIdx, toKvIdx));
+        require(success, "failed to removeRaw");
+    }
+
+    function unmaskChunkWithEthash(
+        uint64 chunkIdx,
+        bytes32 kvHash,
+        address miner,
+        bytes memory maskedChunk
+    ) public view virtual returns (bytes memory) {
+        (bool success, bytes memory unmaskedChunk) = address(sstoragePisaUnmaskDaggerData).staticcall(
+            abi.encode(1, chunkIdx, kvHash, miner, maskedChunk)
```

Comment:
Two places need to be clarified:
1. Ethash type is 1 or 2?
2. The kvHash is 24bytes which is different with the real data hash, and this will lead to incorrect encode key

Reply:
It seems that we have not decided to use high-24byte or low-24byte

Reply:
I believe either works. As long as we pick one convention and use it consistently( everywhere), we should be good. 
Since standard lib function `bytes24` truncates high 24 bytes, I use this fashion. You can pick low 24 bytes if that's more comfortable for you.

Reply:
Good advice.
I think we can also directly use solidity to do this.
`bytes32(uint256(uint192(data)));`
As the result, we will get the low-bytes32 data.